ui/pages/confirmations/components/confirm/info/approve/edit-spending-cap-modal/edit-spending-cap-modal.tsx

```diff
@@ -0,0 +1,165 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React, { useCallback, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { calcTokenAmount } from '../../../../../../../../shared/lib/transactions-controller-utils';
+import { hexToDecimal } from '../../../../../../../../shared/modules/conversion.utils';
+import {
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  TextField,
+  TextFieldType,
+} from '../../../../../../../components/component-library';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import {
+  estimateGas,
+  updateEditableParams,
+} from '../../../../../../../store/actions';
+import { getCustomTxParamsData } from '../../../../../confirm-approve/confirm-approve.util';
+import { useConfirmContext } from '../../../../../context/confirm';
+import { useAssetDetails } from '../../../../../hooks/useAssetDetails';
+import { useApproveTokenSimulation } from '../hooks/use-approve-token-simulation';
+
+export const EditSpendingCapModal = ({
+  isOpenEditSpendingCapModal,
+  setIsOpenEditSpendingCapModal,
+}: {
+  isOpenEditSpendingCapModal: boolean;
+  setIsOpenEditSpendingCapModal: (newValue: boolean) => void;
+}) => {
+  const t = useI18nContext();
+
+  const dispatch = useDispatch();
+
+  const { currentConfirmation: transactionMeta } = useConfirmContext() as {
+    currentConfirmation: TransactionMeta;
+  };
+
+  const { userBalance, tokenSymbol, decimals } = useAssetDetails(
+    transactionMeta.txParams.to,
+    transactionMeta.txParams.from,
+    transactionMeta.txParams.data,
+  );
+
+  const accountBalance = Number(
+    calcTokenAmount(userBalance || '0', Number(decimals || '0')),
+  );
```

Comment:
```suggestion
  const accountBalance = calcTokenAmount(userBalance ?? '0', decimals ?? '0').toFixed();
```

similar to the comment https://github.com/MetaMask/metamask-extension/pull/26845#discussion_r1754570299, we want to avoid transforming the value into a number to preserve the precision. Instead we can use `.toFixed()`. This can be applied to the other accountBalance instances as well. 

for `decimals ?? '0'`, It might be a good idea for us to do this within the method given we are doing this for a few other uses in this PR